syntax = "proto3";

message Optimizer {
  float learning_rate = 1;
  float learning_decay = 2;
}

message EstimatorConfig {
  int32 save_summary_steps = 1;
  int32 save_checkpoints_secs = 2;
  int32 keep_checkpoint_max = 3;
  int32 log_step_count_steps = 4;
}

message Storage {
  string datasets_folder = 1;
  string logs = 2;
}

message Training {
  int32 batch_size = 1;
  int32 num_epochs = 2;
  int32 shuffle_size = 3;
  int32 prefetch_size = 4;
  int32 max_steps = 5;
}

enum Datasets {
  NTU_RGBD = 0;
  MONTALBANOV2 = 1;
}

message Shape {
  int32 width = 1;
  int32 height = 2;
}

message Telegram {
  string token = 1;
  int32 id = 2;
}

enum ActivationFunction {
  RELU = 0;
  SOFTMAX = 1;
}


message DenseLayer {
  message Dropout {
    float rate = 1;
  }
  uint32 units = 1;
  ActivationFunction activation = 2;
  Dropout dropout = 3;
}

message SkeletonNetOptions {
  Optimizer optimizer = 1;
  EstimatorConfig estimator = 2;
  Training training = 3;
  Storage storage = 4;
  Datasets dataset = 5;
  Shape input_shape = 6;
}

message GestureLocalizationOptions {
  Optimizer optimizer = 1;
  EstimatorConfig estimator = 2;
  Training training = 3;
  Storage storage = 4;
  Datasets dataset = 5;
  Telegram telegram = 6;
  repeated DenseLayer hidden_layers = 7;
}